### PR TITLE
Web switch company warning imp detection

### PR DESCRIPTION
**Pull request by hparfr**
Display a warning when you logout/login in another tab or window.

### CODE REVIEW

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 1]
```diff
@@ -33,7 +33,11 @@ openerp.web_switch_company_warning = function (instance) {
             });
 
             w.port.start();
-            w.port.postMessage(this.session.company_id);
+            w.port.postMessage(signature(this.session));
+
+            function signature(session) {
+                return [session.db, session.uid, session.company_id].join();
```
Comment (petrus-v) [inline web_switch_company_warning/static/src/js/switch_company_warning.js:39]: Can you move this function outside of the init method to easily overwrite it on customer project module if needs?

**Comment by petrus-v** [comment]
Hi,

I'm just wondering is this for user convenience or technical limitation, as far as I know there are no limitation to login with 2 different user or on 2 different database?

regards

File: web_switch_company_warning/static/src/js/switch_company_warning.js (hunk 1 above)
Comment (hparfr) [inline web_switch_company_warning/static/src/js/switch_company_warning.js:39]: Ok

**Comment by petrus-v** [comment]
Thanks for this contribution!

:+1:  (code review only!)

**Comment by hparfr** [comment]
> I'm just wondering is this for user convenience or technical limitation, as far as I know there are no limitation to login with 2 different user or on 2 different database?

It's a technical limitation.
As far as I know, you have only one session per host:port

I just tested it now and didn't succeed to maintain 2 sessions on differents db for the same host:port.

PS: I did commit with a wrong email, OCA CLAbot didn't recognize my account.

Thanks for the review petrus-v

**Comment by hparfr** [comment]
I did squash the commit with the wrong email.
Should be ok from the CLA perspective.

**Comment by bealdav** [comment]
tested on Runbot, OK, :+1: 
thanks

**Comment by OSguard** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
CLA checked and confirmed. Tested on runbot and this change works nicely. Nice feature BTW (I had not noticed the module before). :+1:

**Pull request merged**